icu4j/main/core/src/main/java/com/ibm/icu/segmenter/BoundaryIteratorOfInts.java

```diff
@@ -50,9 +50,8 @@ class BoundaryIteratorOfInts {
             0 <= startIdx
                 && startIdx <= sourceLength
                 && breakIter.isBoundary(startIdx);
-        int backFromIdx = isOnBoundary ? startIdx + 1 : startIdx;
 
-        currIdx = breakIter.preceding(backFromIdx);
+        currIdx = isOnBoundary ? startIdx : breakIter.preceding(startIdx);
```

Comment:
why not
```
currIdx = isOnBoundary ? startIdx : breakIter.previous();
```

?

Reply:
I don't think it's more readable, and everywhere in this implementation we need to know intimately how the BreakIterator works. I prefer making this as succinct as in segmentAt(i) -- we know that isBoundary() returned false and moved to the following boundary, and previous() will get us back from there to where we want to be. But I don't know if it actually saves anything. Either way we hopefully have at least the two surrounding boundaries cached. Ok either way; maybe just a comment that we *could* call previous().

Reply:
Fair points. I went ahead and tried to implement your suggestion, but I couldn't because a test broke. The breaking test is `SegmentsTest.testBoundariesBackFrom` with the  test case of the starting index being after the last boundary. `BreakIterator.previous()` returned `DONE` in that test case (leading ultimately to an empty Stream), whereas `BreakIterator.preceding(i)` returns `BreakIterator.last()` when `i` is after the last boundary. So I only updated comments on the existing code, which includes these details.